scripts/solution_3.r

```diff
@@ -0,0 +1,150 @@
+# Assignment 1:  
+library(tweedie) 
+library(ggplot2)
+library(doParallel)
+
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+
+
+# Assignment 2:  
+
+# Parallel Computation within the function
+MTweedieTests <-  
+  function(N,M,sig){ 
+    Cores <- min(parallel::detectCores(), 2)
+    cl <- makeCluster(Cores)
+    registerDoParallel(cl)
+    clusterExport(cl, c("N", "sig"))
+    one_simulation <- function() {
+      replicate(M, simTweedieTest(N)) < sig
+    }
+    results <- parLapply(cl, 1:M, function(i) one_simulation())
+    percent <- sum(unlist(results))/M
+    stopCluster(cl)
+    return(percent)
+  } 
+
+
+# Assignment 3:  
+df <-  
+  expand.grid( 
+    N = c(10,100,1000,5000, 10000), 
+    M = 1000, 
+    share_reject = NA) 
+
+# Parallel Computing
+Cores <- min(parallel::detectCores(), 2)
+
+cl <- makeCluster(Cores)
+
+registerDoParallel(cl)
+
+foreach(i = 1:nrow(df), 
```

Comment:
Since you already divide the simulations in each row into multiple cores, you cannot divide each row into it's own core in addition.